pycbc/events/single.py

```diff
@@ -1,61 +1,141 @@
 """ utilities for assigning FAR to single detector triggers
 """
-from pycbc.events import ranking
+import h5py
+import numpy as np
+from pycbc.events import ranking, trigger_fits as fits
 from pycbc.types import MultiDetOptionAction
+from pycbc import conversions as conv
+from pycbc import bin_utils
 
-class LiveSingleFarThreshold(object):
+class LiveSingle(object):
     def __init__(self, ifo,
                  newsnr_threshold=10.0,
                  reduced_chisq_threshold=5,
                  duration_threshold=0,
-                 fixed_ifar=0):
+                 fit_bins=None,
+                 fit_coeffs=None,
+                 fit_rates=None,
+                 fixed_ifar=None,
+                 fit_thresh=None):
         self.ifo = ifo
-        self.newsnr_threshold = newsnr_threshold
-        self.reduced_chisq_threshold = reduced_chisq_threshold
-        self.fixed_ifar = fixed_ifar
-        self.duration_threshold = duration_threshold
+        self.thresholds = {
+            "newsnr": newsnr_threshold,
+            "reduced_chisq": reduced_chisq_threshold,
+            "duration": duration_threshold}
+        self.fit_info = {
+            "fixed_ifar": fixed_ifar,
+            "bins": fit_bins,
+            "rates": fit_rates,
+            "coeffs": fit_coeffs,
+            "thresh": fit_thresh}
 
     @staticmethod
     def insert_args(parser):
         parser.add_argument('--single-newsnr-threshold', nargs='+',
                             type=float, action=MultiDetOptionAction)
         parser.add_argument('--single-reduced-chisq-threshold', nargs='+',
                             type=float, action=MultiDetOptionAction)
-        parser.add_argument('--single-fixed-ifar', nargs='+',
+        parser.add_argument('--single-fixed-ifar', nargs='+', default=None,
```

Comment:
Args implicitly default to `None`, so you can drop the explicit default setting.